CMakeLists.txt

```diff
@@ -290,7 +290,7 @@ else()
     endif()
 endif()
 
-set_var_to_condition(MIOPEN_USE_COMGR_DEFAULT MIOPEN_EMBED_BUILD)
+set_var_to_condition(MIOPEN_USE_COMGR_DEFAULT (NOT DEFINED MIOPEN_BACKEND_OPENCL) AND (NOT DEFINED MIOPEN_MODE_NOGPU))
```

Comment:
I would prefer to solve this problem in OpenCL and NOGPU handlers
https://github.com/ROCmSoftwarePlatform/MIOpen/blob/72f06aa8cb53c30ac2f87d34378925ff9401f4ed/CMakeLists.txt#L155-L158
https://github.com/ROCmSoftwarePlatform/MIOpen/blob/72f06aa8cb53c30ac2f87d34378925ff9401f4ed/CMakeLists.txt#L223-L226

Reply:
If I do so, then I will need to first set MIOPEN_USE_COMGR_DEFAULT to TRUE and then change it to FALSE in some cases. This makes it harder to track the value of MIOPEN_USE_COMGR_DEFAULT. Do you agree?

Reply:
I suppose you can set MIOPEN_USE_COMGR to FALSE in these two cases, instead of changing MIOPEN_USE_COMGR_DEFAULT

Reply:
MIOPEN_USE_COMGR is an Cmake option, I need separate variable to initialize its default.

Reply:
ohh, yes, right.
Then first case with MIOPEN_USE_COMGR_DEFAULT=true
``If I do so, then I will need to first set MIOPEN_USE_COMGR_DEFAULT to TRUE and then change it to FALSE in some cases. This makes it harder to track the value of MIOPEN_USE_COMGR_DEFAULT. Do you agree?``

Reply:
I want this, just not to lose the reason for these exceptions in the code

Reply:
But the reason is the same - it doesn't work with OpenCL BE and we do not need to for NOGPU ;)

Reply:
I mean that the reason is programmatically expressed in the code. Either in some local place or scattered around. I would prefer locality.

Let's look at it differently. If this is written in C++, then single initialization of MIOPEN_USE_COMGR_DEFAULT allows defining it  as `const`. But with IFs this is not possible.

Reply:
For me local place - in backend handlers, afther parametrs check and error message.
But these is only my preferences. I have already approved this PR.